File: Python/6 - kyu/6 kyu - Highest Scoring Word.py
# https://www.codewars.com/kata/highest-scoring-word/train/python
# My solution
def high(x):
    m,string = 0, ""
    for s in x.split():
        calc = sum(ord(ch.lower()) - 96 for ch in s)
        if calc > m:
            m = calc
            string = s
    return string

# ...
def high(x):
    return max(x.split(), key=lambda k: sum(ord(c) - 96 for c in k))

# ...
def high(words):
    return max(words.split(), key=lambda word: sum(ord(c) - ord('a') + 1 for c in word.lower()))

# ...
def high(s):
    return max( (sum(ord(c) - 96 for c in x), x) for x in s.split() )[1]

# ...

File: C/6 kyu/6 kyu - Highest Scoring Word.c
// https://www.codewars.com/kata/highest-scoring-word/train/c
// My solution
#include <string.h>

static int scoreWord(char* word) {
    int score = 0;
    for ( char* letter = word ; *letter ; letter++ ) {
        score += *letter - 'a' + 1;
    }
    return score;
}

char  *highestScoringWord(const char *str)
{
    char *copy = strdup( str );
    
    char *token = strtok( copy, " " );
    char *result = strdup( token );  // Malloc
    int highScore = scoreWord( result );
    
    while( token != NULL ) {
        int newScore = scoreWord( token );
        if ( newScore > highScore ) {
            free( result );  // Free
            result = strdup( token ); // Malloc
            highScore = newScore;
        }
        token = strtok( NULL, " " );
    }
    free( copy );
    return result;
}